src/devices/Mhz19b/Mhz19b.cs

```diff
@@ -0,0 +1,218 @@
+        }
+
+        /// <summary>
+        /// Gets the current CO2 concentration from the sensor.
+        /// The validity is true if the current concentration was successfully read.
+        /// If the serial communication timed out or the checksum was invalid the validity is false.
+        /// If the validity is false the ratio is set to 0.
+        /// </summary>
+        /// <returns>CO2 concentration in ppm and validity</returns>
+        public (Ratio, bool) GetCo2Reading()
+        {
+            Ratio concentration = Ratio.FromPartsPerMillion(0);
+            bool validity = false;
+
+            try
+            {
+                var request = CreateRequest(Command.ReadCo2Concentration);
+                request[(int)MessageFormat.Checksum] = Checksum(request);
+
+                _serialPort.Open();
+                _serialPort.Write(request, 0, request.Length);
+
+                byte[] response = new byte[MessageSize];
+                if ((_serialPort.Read(response, 0, response.Length) == response.Length) && (response[(int)MessageFormat.Checksum] == Checksum(response)))
+                {
+                    concentration = Ratio.FromPartsPerMillion((int)response[(int)MessageFormat.DataHigh] * 256 + (int)response[(int)MessageFormat.DataLow]);
+                    validity = true;
+                }
+            }
+            finally
+            {
+                _serialPort.Close();
+            }
+
+            return (concentration, validity);
+        }
+
+        /// <summary>
+        /// Initiates a zero point calibration.
+        /// The sensor doesn't respond anything, so this is fire and forget.
+        /// </summary>
+        /// <returns>true, if the command could be send</returns>
+        public bool ZeroPointCalibration() => SendRequest(CreateRequest(Command.CalibrateZeroPoint));
+
+        /// <summary>
+        /// Initiate a span point calibration.
+        /// The sensor doesn't respond anything, so this is fire and forget.
+        /// </summary>
+        /// <param name="span">span value, e.g. 2000[ppm]</param>
+        /// <returns>true, if the command could be send</returns>///
+        public bool SpanPointCalibration(int span)
+        {
+            var request = CreateRequest(Command.CalibrateSpanPoint);
+            // set span in request, c. f. datasheet rev. 1.0, pg. 8 for details
+            request[(int)MessageFormat.DataHigh] = (byte)(span / 256);
+            request[(int)MessageFormat.DataLow] = (byte)(span % 256);
+
+            return SendRequest(request);
+        }
+
+        /// <summary>
+        /// Switches the autmatic baseline correction on and off.
+        /// The sensor doesn't respond anything, so this is fire and forget.
+        /// </summary>
+        /// <param name="state">State of automatic correction</param>
+        /// <returns>true, if the command could be send</returns>
+        public bool AutomaticBaselineCorrection(AbmState state)
```

Comment:
Would it be easier if you are just using a property? 
AbmState is only on and off. That will make it simpler
I understand SendRequest returns a potential status but you can as well raise an IOException as if you have a problem with the serial port, none of the commands will work anyway. 
If there are good reasons why not having a bool property (or if you want to keep the enum a AbmState property), then please use a verb for the function like SetAutomaticBaselineCorrection or SwitchAutomaticBaselineCorrection

Reply:
I wouldn't prefer a property over a function here, as the call requires a certain execution time. From my understanding a property access should be rather instantly be finished. In case everything is fine this would be (almost) the case, but if there is an issue with the serial port it takes until the timeout happens. Even if this would be reduced to, let's say 10 or 100ms, this isn't the timing I expect from a property. Furthermore there is no chance in reading back the current actual setting, therefore the property wouldn't be readable in the first place. 

Whether I use an exception or a return code for reporting the status is a matter of taste or project coding guideline.  
Usually I prefer the return code, but if preferred by the iot-project I could change this. Please advice.

Reply:
In that case, as it's related to the SerialPort itsef, I would prefer an Exception. 
In the case of reading in a Try pattern, it's of course find to return a bool.
So ok then for the function, that said, you'll still need to rename it with an action verb like SetAutomaticBaselineCorrection or SwitchAutomaticBaselineCorrection

Reply:
btw, for the naming convention and Try pattern, see https://github.com/dotnet/iot/tree/master/src/devices#naming-of-functions

Reply:
Done (renamed method to SetAutomaticBaselineCorrection)